geniza/corpus/models.py

```diff
@@ -144,6 +147,18 @@ def iiif_thumbnails(self):
             for i, img in enumerate(images)
         ))
 
+    def save(self, *args, **kwargs):
+        '''Remember how shelfmarks have changed by keeping a semi-colon list 
+        in the old_shelfmarks field'''
+        if self.pk and self.has_changed('shelfmark'):
+            if self.old_shelfmarks:
+                old_shelfmarks = set(self.old_shelfmarks.split(';'))
+                old_shelfmarks.add(self.initial_value('shelfmark'))
+                self.old_shelfmarks = ';'.join([sm for sm in old_shelfmarks if sm != self.shelfmark])
```

Comment:
random thought: could you do `';'.join(old_shelfmarks - {self.shelfmark})`? seems a little clearer and might be slightly faster

Reply:
I don't know about faster, but it certainly is more elegant! I'll commit this and merge